python/ray/data/_internal/datasource/hudi_datasource.py

```diff
@@ -64,12 +112,20 @@ def _perform_read(
                 input_files.append(full_path)
                 size_bytes += file_slice.base_file_size
 
-            metadata = BlockMetadata(
-                num_rows=num_rows,
-                input_files=input_files,
-                size_bytes=size_bytes,
-                exec_stats=None,
-            )
+            if self._query_type == HudiQueryType.SNAPSHOT:
+                metadata = BlockMetadata(
+                    num_rows=num_rows,
+                    input_files=input_files,
+                    size_bytes=size_bytes,
+                    exec_stats=None,
+                )
+            elif self._query_type == HudiQueryType.INCREMENTAL:
+                metadata = BlockMetadata(
+                    num_rows=None,
+                    input_files=input_files,
+                    size_bytes=None,
+                    exec_stats=None,
+                )
```

Comment:
This is needed because of https://github.com/apache/hudi-rs/issues/401, which will be fixed in the next release.

Reply:
@alexeykudinkin I can add this as in-code comment too if you like. please review the rest part so i can address any other comments altogether.